### PR TITLE
fix(k8s-idpe-28726): use backoff on startup authz probe()

### PR DESCRIPTION
**Pull request by wiedld**
Closes https://github.com/influxdata/k8s-idpe/issues/28726

Authz sidecar may come online later than the routers and ingesters. These services will attempt to probe for the authz service, and fail when no connection is available.

During the startup check only (`probe()`) enable backoff and retry. 
First commit (the test) intentionally fails, prior to the fix.

---
* test(k8s-idpe-28726): test authz probe() scenario where the tcp connection response is delayed (f4a41df3e)

* fix(k8s-idpe-28726): provide backoff and retry logic on authz probe (0b88af823)
     ```
      Note that an error returned from authz is still a valid response,
      therefore we are using a conditional retry based on communication errors.
     ```

## Checklist
- [x] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [x] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: authz/src/authorizer.rs [hunk 1]
```diff
@@ -34,13 +37,25 @@ pub trait Authorizer: std::fmt::Debug + Send + Sync {
...
+            .retry_with_backoff("probe iox-authz service", move || {
+                async {
+                    match self.permissions(Some(b"".to_vec()), &[]).await {
+                        // got response from authorizer server
+                        Ok(_)
+                        | Err(Error::Forbidden)
+                        | Err(Error::InvalidToken)
+                        | Err(Error::NoToken) => ControlFlow::Break(Ok(())),
+                        // communication error == Error::Verification
+                        Err(e) => ControlFlow::<_, Error>::Continue(e),
```
Comment (wiedld) [inline authz/src/authorizer.rs:53]: @mhilton -- can I get a confirmation/correction that the `Error::Verification` is always a communication error?

File: authz/src/authorizer.rs [hunk 2]
```diff
@@ -34,13 +37,25 @@ pub trait Authorizer: std::fmt::Debug + Send + Sync {
...
+    ///
+    /// Test is performed during deployment, with ordering of availability not being guaranteed.
     async fn probe(&self) -> Result<(), Error> {
-        match self.permissions(Some(b"".to_vec()), &[]).await {
-            // got response from authorizer server
-            Ok(_) | Err(Error::Forbidden) | Err(Error::InvalidToken) => Ok(()),
-            // other errors, including Error::Verification
-            Err(e) => Err(e),
-        }
+        Backoff::new(&BackoffConfig::default())
```
Comment (wiedld) [inline authz/src/authorizer.rs:43]: @cannonpalms -- do you mind providing a recommendation for what [backoff settings](https://github.com/influxdata/influxdb_iox/blob/98960a353cefd0a5d58d3089408ccb5f08815925/backoff/src/lib.rs#L31) we should use?

Comment (cannonpalms) [reply authz/src/authorizer.rs:43]: After review, the sidecar we're waiting on starts up in a few hundred milliseconds in the healthy case. The [default](https://github.com/influxdata/influxdb_iox/blob/98960a353cefd0a5d58d3089408ccb5f08815925/backoff/src/lib.rs#L48-L51) configuration that you're using now looks to be exactly what's needed. 

In the majority of cases where a retry is necessary, we will succeed on the first retry.

Comment (wiedld) [reply authz/src/authorizer.rs:43]: Perfect. TY!

File: authz/src/authorizer.rs (hunk 1 above)
Comment (mhilton) [reply authz/src/authorizer.rs:53]: I don't completely remember, but I'm pretty sure that was the intention.

**Review by mhilton — APPROVED** [review]

**Pull request merged**